### DISCUSSION TITLE
$page->image gak bisa di resize satu-satu seperti ini ya kang?

### DISCUSSION THREAD

**Discussion by yuliakbard** [mecha-cms/mecha#222] [Q&A]
Kenapa hasil untuk resize url gambar dibawah ini, gak bisa keluar sesuai dengan yang kita tulis kang?
```
A = <?=$page->image(360,360,100); ?><br/> // Output : A = .../lot/asset/.cache/d3/91/19/82/e6/b0/eb/7b/81/2d/e5/84/12/33/d4/80.jpg
B = <?=$page->image(180,180,100); ?><br/> // Output : A = .../lot/asset/.cache/d3/91/19/82/e6/b0/eb/7b/81/2d/e5/84/12/33/d4/80.jpg
C = <?=$page->image(72,72,100); ?><br/> // Output : A = .../lot/asset/.cache/d3/91/19/82/e6/b0/eb/7b/81/2d/e5/84/12/33/d4/80.jpg
```

**Comment by taufik-nurrohman** [comment]
Emang yang sesuai bagaimana mas?

**Reply by yuliakbard** [reply]
```
A = <?=$page->image(360,360,100); ?><br/> // Output : A = .../lot/asset/.cache/d3/91/19/82/e6/b0/eb/7b/81/2d/e5/84/12/33/d4/80.jpg
B = <?=$page->image(180,180,100); ?><br/> // Output : B = .../lot/asset/.cache/d3/91/19/82/e6/b0/eb/7b/81/2d/e5/84/12/33/d4/80.jpg
C = <?=$page->image(72,72,100); ?><br/> // Output : C = .../lot/asset/.cache/d3/91/19/82/e6/b0/eb/7b/81/2d/e5/84/12/33/d4/80.jpg
```

yang A size 360, yang B size 180, yang C size 72, tetapi yang tampil url size 360 di A, B, C

**Reply by taufik-nurrohman** [reply]
Itu karena nilainya sudah masuk ke properti cache kayaknya mas. Sebelum memanggil gambar yang lain, mungkin perlu dihapus cache tersebut:

~~~ php
echo $page->image(360,360);
$page->set('image', null); // v2.6.4
echo $page->image(180,180);
~~~

**Reply by taufik-nurrohman** [reply]
Sekarang harusnya sudah bisa.

**Reply by yuliakbard** [reply]
kalau dikombinasikan dengan discussion #178 gimana kang?

```
function source_set(string $source, array $set = []) {
    $out = [];
    foreach ($set as $v) {
        $out[] = $source . ' ' . $v . 'w';
    }
    return implode(', ', $out);
}
```

biar bisa dipanggil dengan `<?= self::source_set($page->image, [100,200,300]);?>`

**Reply by yuliakbard** [reply]
**1.** 
`<img srcset="<?=$page->image(72) ." 72w,". $page->image(180) ." 18w," . $page->image(300) ." 300w";?>">
// nilai instrinsic size gambar tidak berubah walau url nya berubah
`

**2.**

```
//sedangkan kalau dibuat satu-satu berfungsi dengan baik, instrinsic size gambar sesuai dengan yang kita render

<img src="<?=$page->image(72);?>">
<img src="<?=$page->image(180);?>">
<img src="<?=$page->image(300);?>">
```

**3.**
```
// kalau seperti ini:
<?php
foreach ([100, 200, 300] as $v) {
    $gambar[] = $page->image($v). ' ' . $v . 'w';
}
    $out = implode(', ', $gambar);
?>

<img srcset="<?= $out; ?>">
// hasilnya sama seperti yang pertama.
```

**Reply by taufik-nurrohman** [reply]
Ukuran intrinsik itu seperti apa mas? Proporsional? Jujur Saya jarang banget pakai fitur HTML5 `srcset`.

**Comment by yuliakbard** [comment]
kalau kita inspect elemen itu ada ukuran gambar yang ditampilkan, sesuai kode gambar yang kita pakai. Misal: `$page->image(72)`. untuk kode php 1 dan 3 itu ukuran yang bakal ditampilkan itu ukuran tertinggi dari yang kita panggil, walaupun secara url `srcset="http://localhost/hero/image/t/**72**/d03a1d00.png 72w, http://localhost/hero/image/t/**180**/d03a1d00.png 180w, http://localhost/hero/image/t/**300**/d03a1d00.png"` ukuran yang akan ditampilkan pada nilai instrinsicnya itu yang paling tinggi yaitu 300, meskipun sudah kita hover setiap url nya satu-satu.

`Ini Ini Instrinsic Size gambar yang ane maksud kang`
![Screenshot 2023-03-14 004918](https://user-images.githubusercontent.com/96925342/224788098-2b1e6d60-2a48-4252-b97c-869332c26451.png)

`Ini untuk penggunaan kode nomor 1 dan 3 outputnya sama`
![image](https://user-images.githubusercontent.com/96925342/224789623-81756bf2-a43c-4322-8a00-372ef963c171.png)

kalau ane buat seperti yang nomor 2, itu baru ukuran instrinsic size nya beda beda.

kalau pake statically nilai instrinsic size bakal sesuai dengan yang kita panggil pada kode yang ditulis #178,

**Reply by yuliakbard** [reply]
harapannya sih di mecha sourceset ini berfungsi seperti yang ane harapkan, karena dari jaman blogger pun, ane pake fungsi sourceset buat nampilin gambarnya. 

untuk sekarang ane masih ada alternatif buat nampilin gambar untuk sourceset nya masih pake cara yang di #178

**Reply by yuliakbard** [reply]
tapi ane gak tau juga nih sekarang, soalnya role model website yang ane jadi patokan itu ngikutin kodingan html nya instagram/ facebook untuk pemuatan gambarnya. hehehe. kayaknya instagram dan facebook udah gak pake sourceset lagi 😅
